lib/CAN/Can.cpp

```diff
@@ -254,20 +254,20 @@ Status CANBus::write(uint16_t message_id, const uint8_t data[], uint32_t size)
         // mask for the first open transmission mailbox
         const uint32_t transmit_mailbox = (hcan->TSR & (0b11ul << CAN_TSR_CODE)) >> 24;
 
-        hcan->sTxMailBox[transmit_mailbox].TIR = (message_id << CAN_TI0R_STID) | CAN_RTR_DATA;
-        hcan->sTxMailBox[transmit_mailbox].TDTR = 8; // always only transmit 8 bytes for simplicity
+        hcan->sTxMailBox[transmit_mailbox].TIR = (message.id << CAN_TI0R_STID) | CAN_RTR_DATA;
+        hcan->sTxMailBox[transmit_mailbox].TDTR = message.size; // always only transmit 8 bytes for simplicity
```

Comment:
If you can confirm to me this will always be 8, it's fine

Reply:
It's not, I should've removed the comment. Will do before submitting.

Because we're using a fixed size array in the message struct, we can guarantee that accessing any of the 8 bytes is safe. We will only ever transmit the lower N bytes (N = TDTR = message.size).